packages/nectar/src/sagas/messages/updateMessagesSendingStatus/updateMessagesSendingStatus.saga.test.ts

```diff
@@ -0,0 +1,60 @@
+import { setupCrypto } from '@quiet/identity'
+import { Store } from '../../store.types'
+import { prepareStore } from '../../../utils/tests/prepareStore'
+import { getFactory } from '../../..'
+import { FactoryGirl } from 'factory-girl'
+import { combineReducers } from 'redux'
+import { reducers } from '../../reducers'
+import { expectSaga } from 'redux-saga-test-plan'
+import { Identity } from '../../identity/identity.types'
+import { identityActions } from '../../identity/identity.slice'
+import { updateMessagesSendingStatusSaga } from './updateMessagesSendingStatus.saga'
+import { communitiesActions, Community } from '../../communities/communities.slice'
+import { publicChannelsActions } from '../../publicChannels/publicChannels.slice'
+import { ChannelMessage } from '../../publicChannels/publicChannels.types'
+import { messagesActions } from '../messages.slice'
+
+describe('updateMessagesSendingStatusSaga', () => {
+  let store: Store
+  let factory: FactoryGirl
+
+  let community: Community
+  let alice: Identity
+  let message: ChannelMessage
+
+  beforeAll(async () => {
+    setupCrypto()
+
+    store = prepareStore().store
+    factory = await getFactory(store)
+
+    community = await factory.create<
+    ReturnType<typeof communitiesActions.addNewCommunity>['payload']
+    >('Community')
+
+    alice = await factory.create<ReturnType<typeof identityActions.addNewIdentity>['payload']>(
+      'Identity',
+      { id: community.id, nickname: 'alice' }
+    )
+
+    message = (await factory.build<typeof publicChannelsActions.test_message>(
+      'Message',
+      { identity: alice }
+    )).payload.message
+  })
+
+  test('remove pending message status', async () => {
+    const reducer = combineReducers(reducers)
+    await expectSaga(
+      updateMessagesSendingStatusSaga,
+      publicChannelsActions.incomingMessages({
```

Comment:
Is it confusing that we're adding these messages to a pile called incoming messages? Or is that not what's happening here?

Reply:
It's only calling the action that updates redux store. It's important to do it this way here as this action triggers the exact saga we want to test. In real life this action is being called by the socket event that comes from waggle (but in this kind of tests we don't have waggle running)

Reply:
It's just an argument provided to the saga we're testing

Reply:
This argument comes in the form of redux's action creator